pyvrp/Model.py

```diff
@@ -35,12 +41,50 @@ def __init__(
         distance: int,
         duration: int,
     ):
+        if distance < 0 or duration < 0:
+            raise ValueError("Cannot have negative edge distance or duration.")
+
+        if frm == to and (distance != 0 or duration != 0):
+            raise ValueError("A self loop must have 0 distance and duration.")
+
+        if max(distance, duration) > MAX_VALUE:
+            msg = """
+            The given distance or duration value is very large. This may impact
+            numerical stability. Consider rescaling your input data.
+            """
+            warn(msg, ScalingWarning)
+
         self.frm = frm
         self.to = to
         self.distance = distance
         self.duration = duration
 
 
+class Profile:
+    """
+    Stores a routing profile.
```

Comment:
I think this is a really big difference between academia and industry, because this is the terminology I've found online quite a bit. For example over [here](https://reinterpretcat.github.io/vrp/concepts/pragmatic/routing/profile.html), [here](https://docs.mapbox.com/help/glossary/routing-profile/), and [here](https://wiki.openstreetmap.org/wiki/Routing_profiles). A full description should probably go into the docs at some point, but we do not have algorithmic support for this so there's no point writing that yet. I'll add a little more to this class for now.